cachepool/cachepool.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python2
+# coding: utf-8
+
+import copy
+import Queue
+import logging
+
+logger = logging.getLogger(__name__)
+
+class CachePoolError(Exception): pass
+class CachePoolGeneratorError(CachePoolError): pass
+class CachePoolCloseCallbackError(CachePoolError): pass
+
+class CachePoolReuseDeciderError(CachePoolError): pass
+
+class CachePool(object):
+
+    def __init__(self,
+                    generator,
+                    generator_args=[],
```

Comment:
Never use a reference-type object as default argument: every time this function is called, the same instance of `[]` or `{}` will be used.
Use primitive type instead: bool, int or None